### PR TITLE
Kp/ordered collections of data

### PR DESCRIPTION
**Pull request by kp483**
Previously:
- Vocab at top of intro lesson
- Intro lesson content pulled from C16 branch

Now:
- Vocab in separate lesson at the end (I feel like it's very long and bulky and introduces way too much; I think it's better as a review resource)
    - Will review definitions in separate PR
- Grammar and language revised for intro lesson
- Walkthrough of time complexity for array and linked list operations marked as review
- Callout added to highlight difference between binary search and binary search trees
- Lesson order reorganized to accommodate vocab lesson

For review:
- Is restructuring of vocab to separate lesson good?
- Grammar/clarity of language review

### CODE REVIEW

File: 03-Binary-Search-Trees/01-Ordered-Collections-Of-Data.md [hunk 1]
```diff
@@ -23,23 +23,6 @@ Students should be able to:
 - [Slide Deck](https://docs.google.com/presentation/d/1Fj0deIUswGZ3ooJMpgVUqPEaWHKTkQ1w2Ci-yf8v66M/edit#slide=id.p)
 - [BST Exercise](https://github.com/Ada-C16/tree-practice)
```
Comment (char-adadev) [inline 03-Binary-Search-Trees/01-Ordered-Collections-Of-Data.md:24]: Is it ok for us to link to the C16 project here?

File: 03-Binary-Search-Trees/04-Binary-Search-Trees-Vocab.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+## Vocabulary and Synonyms
```
Comment (char-adadev) [inline 03-Binary-Search-Trees/04-Binary-Search-Trees-Vocab.md:1]: Good move!

**Review by char-adadev — APPROVED** [review]
Thank you so much for working on this! Great work 💪🏾 

I had one thing I wanted to point out, but nothing major. There seems to be some duplicate info in the first two lessons -> 01-Ordered-Collections-Of-Data.md & 02-Binary-Search-Trees.md.

I think this is unindentional, but I wanted to double check. They also have the same name in Learn:
Screen Shot 2022-08-31 at 12 46 34 PM

Approving for now as I think it'll be a pretty quick follow-up for that duplicated material.

File: 03-Binary-Search-Trees/01-Ordered-Collections-Of-Data.md (hunk 1 above)
Comment (kp483) [reply 03-Binary-Search-Trees/01-Ordered-Collections-Of-Data.md:24]: Will make a note to go back and change once the C17 project exists

**Comment by kp483** [comment]
> Thank you so much for working on this! Great work 💪🏾
> 
> I had one thing I wanted to point out, but nothing major. There seems to be some duplicate info in the first two lessons -> 01-Ordered-Collections-Of-Data.md & 02-Binary-Search-Trees.md.
> 
> I think this is unindentional, but I wanted to double check. They also have the same name in Learn: Screen Shot 2022-08-31 at 12 46 34 PM
> 
> Approving for now as I think it'll be a pretty quick follow-up for that duplicated material.

Yep! Should've noted that, but it's been removed as part of the next PR

**Pull request merged**